### ISSUE TITLE
SharePoint Hosted app is not installing or upgrading

### ISSUE THREAD

**Issue by tmax-siva** [SharePoint/sp-dev-docs#9645]
### Target SharePoint environment

other (enter in the "Additional environment details" area below)

### What SharePoint development model, framework, SDK or API is this about?

SharePoint Add-ins

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [x] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version (Version 123.0.6312.123 (Official Build) (64-bit))
- Visual studio (version 2019 Community version)

### Describe the bug / error

We've had a SharePoint hosted app project running live for the past year and a half. Recently, we aimed to deploy a new version of the app by replacing the current version in the tenant-level app catalog. However, when we attempted to upgrade the app on a specific site where it was already installed, we encountered an upgrade error. Additionally, when we tried to install the new version on entirely new sites, we faced installation errors.

Even after reverting all changes and attempting to redeploy with a different app version, we still encountered installation issues. Below are the errors we encountered during both the upgrade and fresh installation attempts.

Installation error:
![image](https://github.com/SharePoint/sp-dev-docs/assets/81231059/05dbd94b-2ecd-45f3-9fa9-7b1c9806298f)

Upgrade error:
![image](https://github.com/SharePoint/sp-dev-docs/assets/81231059/e68f8af7-66a5-4f6d-b01b-c4d6392ab456)

We don't have any document libraries or files with unsupported characters or symbols.

We would greatly appreciate any assistance in resolving this issue. Thank you in advance for your help.

### Steps to reproduce

1.Take hosted app which is already deployed or all new hosted app
2. Deploy in tenant level app catalog
3. Open specific team site and install the add in freshly or upgrade the existing one, now it will throw an error

Tried this in both developer site and team site, in all the places it is giving me same error.

4 months before we deployed with some changes, it was upgraded and deployed successfully without any issues.

### Expected behavior

Expected behavior is, without any issue on installation or upgrade it should be added or upgraded in specific site.

**Comment by mkm17** [comment]
Hi @tmax-siva,  Did you try to deploy the new version of the SharePoint hosted app on SharePoint Online? If so, maybe the issue is related to the announced retirement process described here https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/retirement-announcement-for-add-ins. 

The only date which could be related to your case is this one 
![image](https://github.com/SharePoint/sp-dev-docs/assets/17220108/d600457b-469a-4932-ae10-b04d135125c2)

but as I understand you tried to update the solution by appcatalog and this process should be still possible until April 2026.
https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/add-ins-and-azure-acs-retirements-faq#will-app-catalogs-stay .

**Comment by tmax-siva** [comment]
Yes, as you rightly said am trying to upgrade or install through app catalog only, then too the issue is coming. Because of this we are not able to deploy a important change to client tenant.

**Comment by mkm17** [comment]
@tmax-siva I'm sorry, if this problem isn't related to the retirement process, then I'm not sure what could be the reason here. Perhaps someone from support would have more insights on this issue.

**Comment by Amey-MSFT** [comment]
Hello @tmax-siva,
Thank you for bringing this issue to our attention. Could you please confirm if the issue still persists for you?

**Comment by Amey-MSFT** [comment]
We are closing this issue for now. If the problem persists, feel free to reopen it or open a new one.
